src/core/Database.cpp

```diff
@@ -355,3 +320,34 @@ const CompositeKey & Database::key() const
     return m_data.key;
 }
 
+QVariantMap& Database::kdfParams() {
+    return m_data.kdfParams;
+}
+
+void Database::setKdfParams(QVariantMap params) {
```

Comment:
The Qt docs say, you can pass them by value, but I'm not totally convinced. Even though the dynamically allocated data are not copied, the actual container object is still more than just a pointer. Passing objects by const reference IMHO also leads to cleaner programming, because you make a decision whether a parameter will be modified or not.